### PR TITLE
move original getCmd back

### PR DESCRIPTION
**Pull request by irisdingbj**

### CODE REVIEW

**Comment by ozevren** [comment]
This is deprecated, in favor of kubectl. I am not sure why we would want to resurrect?

**Comment by irisdingbj** [comment]
> This is deprecated, in favor of kubectl. I am not sure why we would want to resurrect?
@ozevren  ```istioctl get``` will put more info than ```kubectl get ``` so we want to bring this command back, an example as below:
```
DESTINATION-RULE NAME   HOST           SUBSETS                      NAMESPACE   AGE
reviews                 reviews        v1,v2,v3                     default     7d
ratings                 ratings        v1,v2,v2-mysql,v2-mysql-vm   default     7d
productpage             productpage    v1                           default     7d
iris                    iristest.com                                default     16d
details                 details        v1,v2                        default     7d```

**Comment by howardjohn** [comment]
@irisdingbj we recently changed kubectl to show some more info, can't we just modify it further to expose all the information we want? or is what kubectl offers not flexible enough?

**Comment by esnible** [comment]
@howardjohn Two problems with _kubectl_.  The first is that folks like `istioctl get all`, and `kubectl` doesn't have it.

(With _kubectl_ you can do `kubectl get gw,vs,dr,se`.  I have an incantation to get ALL the Istio types, but it is frustratingly complex:

```
ISTIO_CRDS=$(kubectl get crds -o jsonpath="{range $.items[?(@.spec.names.categories[0]=='istio-io')]}{.spec.names.plural},{end}")
ISTIO_CRDS=${ISTIO_CRDS%\,}  # Remove trailing comma
kubectl get $ISTIO_CRDS -o yaml --all-namespaces
```
)

The second problem is custom columns.  I added them to _kubectl_ with https://github.com/istio/istio/issues/11552 .  Unfortunately _kubectl_ cannot handle full JSONpath with `[*]`.  So we can't have a kubectl column showing `.spec.subsets[*].name`.

I wish I knew if and when _kubectl_ would get full JSONPath in server side columns.

File: istioctl/cmd/deprecated_cmd.go [hunk 1]
```diff
@@ -276,9 +276,8 @@ var (
 	}
 
 	getCmd = &cobra.Command{
-		Use:        "get <type> [<name>]",
-		Deprecated: "Use `kubectl get` instead (see https://kubernetes.io/docs/tasks/tools/install-kubectl)",
-		Short:      "Retrieve policies and rules",
+		Use:   "get <type> [<name>]",
```
Comment (sdake) [inline istioctl/cmd/deprecated_cmd.go]: if you wish to revert this deprecation, can you identify if this was only reverted in master, or was reverted in a release branch?

Also I think this file is the wrong place to put it.

**Comment by ozevren** [comment]
Keeping overlapping features of kubectl creates a lot of surface area that needs to be tested and maintained. It is also confusing (our online documentation was mixing and matching both profusely, making things more complicated).

IMO, it is better to not have this functionality, and ask for some of the missing features from Kubernetes folks. The things being mentioned here seems like goodness overall for CRDs in general, and not specific to Istio.

**Comment by esnible** [comment]
We are waiting for an implementation of https://github.com/kubernetes/kubectl/issues/517 to provide `kubectl get` with all the columns `istioctl get` provided.

I don't find it hard to type `kubectl get gw,vs,dr,se ...` instead of `istioctl get all ...` but I can understand that beginners might find it hard.  Idea: could we have the bash completion for `istioctl` do something like `export ISTIO_ALL=gw,vs,dr,se,clusterrbacconfigs,envoyfilters,policies,servicerolebindings,serviceroles,sidecars`?  Then users could do `kubectl get $ISTIO_ALL ...`

It was @linsun who suggested restoring this while we wait for improvements to Kubernetes server side printing.  Lin, should we address the concerns on this PR for 1.2 or abandon it and hope Kubernetes fixes https://github.com/kubernetes/kubectl/issues/517 ?  I am worried about waiting for K8s because even if someone submits a suitable PR we might not be able to use it because it would mean a different CRD `additionalPrinterColumns` value for K8s versions.

**Pull request closed without merging**